consensus/cryptarchia-engine/src/lib.rs

```diff
@@ -443,7 +443,7 @@ where
             // We calculate LCA once and store it in `ForkInfo` so it can be consumed
             // elsewhere without the need to re-calculate it.
             let lca = self.branches.lca(&local_chain, &fork);
-            (lca.length <= target_height).then_some(ForkDivergenceInfo { tip: fork, lca })
+            (lca.length < target_height).then_some(ForkDivergenceInfo { tip: fork, lca })
```

Comment:
Can you please also update the function comment? It says that it deletes chains at or before the provided `depth`, but this should change. There is nothing wrong with this function, but rather what `depth` is provided elsewhere, but I agree this change will make it easier to provide a value of `depth` which is simply derived from the height of the LIB.